plugins/experimental/rate_limit/sni_selector.cc

```diff
@@ -139,6 +262,26 @@ SniSelector::setupQueueCont()
     _queue_cont = TSContCreate(sni_queue_cont, TSMutexCreate());
     TSReleaseAssert(_queue_cont);
     TSContDataSet(_queue_cont, this);
-    _action = TSContScheduleEveryOnPool(_queue_cont, QUEUE_DELAY_TIME.count(), TS_THREAD_POOL_TASK);
+    _queue_action = TSContScheduleEveryOnPool(_queue_cont, QUEUE_DELAY_TIME.count(), TS_THREAD_POOL_TASK);
   }
 }
+
+///////////////////////////////////////////////////////////////////////////////
+// Startup of the SNI selector hooks and config reload continuation and instance
+//
+void
+SniSelector::startup()
+{
+  TSCont sni_cont = TSContCreate(sni_limit_cont, nullptr);
+
+  TSReleaseAssert(sni_cont);
+
+  _instance.store(new SniSelector());
+  TSHttpHookAdd(TS_SSL_CLIENT_HELLO_HOOK, sni_cont);
+  TSHttpHookAdd(TS_VCONN_CLOSE_HOOK, sni_cont);
+
+  auto config_cont = TSContCreate(sni_config_cont, TSMutexCreate());
+
+  TSReleaseAssert(config_cont);
+  TSContScheduleEveryOnPool(config_cont, std::chrono::milliseconds{10000}.count(), TS_THREAD_POOL_TASK);
```

Comment:
Is there a reason why we want to change the config file every 10 seconds vs using TSMgmtUpdateRegister() and wait for an explicit config reload 

just to make it more convenient?

Can a race condition happen when the config yaml is only partial updated when the config_cont is scheduled to run and therefore resulting in some inconsistent behavior?

Reply:
Yeh, for convenience, but we can change to be triggered with traffic_ctl.

Reply:
I think it is better to be triggered by traffic_ctl